### ISSUE TITLE
Garbage collection command fails because of msising `gc_units_to_rmfiles`

### ISSUE THREAD

**Issue by con-f-use** [xonsh/xonsh#3131]
Here is my shell session:
```
┌─18-12:44[confus@confusion6 ~]                                                    
└$ xonfig info                                                                  
+------------------+----------------------+
| xonsh            | 0.9.3                |
| Git SHA          | 679d4e06             |
| Commit Date      | May 14 17:45:24 2019 |
| Python           | 3.6.7                |
| PLY              | 3.11                 |
| have readline    | True                 |
| prompt toolkit   | 2.0.9                |
| shell type       | prompt_toolkit2      |
| pygments         | 2.4.0                |
| on posix         | True                 |
| on linux         | True                 |
| distro           | ubuntu               |
| on darwin        | False                |
| on windows       | False                |
| on cygwin        | False                |
| on msys2         | False                |
| is superuser     | False                |
| default encoding | utf-8                |
| xonsh encoding   | utf-8                |
| encoding errors  | surrogateescape      |
+------------------+----------------------+

┌─18-12:46[confus@confusion6 ~]                                                    
└$ history gc                                                                   
Exception in thread Thread-451:
Traceback (most recent call last):
  File "/usr/lib/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/home/jan/.local/lib/python3.6/site-packages/xonsh/history/__amalgam__.py", line 352, in run
    rmfiles_fn = self.gc_units_to_rmfiles.get(units)
AttributeError: 'JsonHistoryGC' object has no attribute 'gc_units_to_rmfiles'
```
I'm not nearly deep enough into xonsh to make sense of it.

**Comment by scopatz** [comment]
Oh interesting.  Thanks for bringing this up.  This seems like a bug. I can reproduce.

**Comment by scopatz** [comment]
potential fix in #3138

**Comment by melund** [comment]
Thanks for reporting this @con-f-use. Please reopen if the issue isn't solved.
